galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.gradle.ubuntu.manager/src/main/java/dev/galasa/gradle/ubuntu/GradleUbuntuInstallation.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright contributors to the Galasa project
+ */
+
+package dev.galasa.gradle.ubuntu;
+
+import dev.galasa.framework.spi.ValidAnnotatedFields;
+import dev.galasa.gradle.GradleVersion;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Represents a Gradle installation on a Ubuntu image.
+ * 
+ * <p>
+ * Used to populate a {@link IGradleUbuntuInstallation} field
+ * </p>
+ * 
+ * @author Matthew Chivers
+ *
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.FIELD })
+@GradleUbuntuManagerField
+@ValidAnnotatedFields({ IGradleUbuntuInstallation.class })
+public @interface GradleUbuntuInstallation {
+    
+    /**
+     * Version of Gradle to be installed.
+     */
+    GradleVersion gradleVersion() default GradleVersion.v7_1_1;
+
+    /**
+     * The tag to be assigned to this Gradle installation.
+     */
+    String gradleTag() default "PRIMARY";
+
+    /**
+     * The tag of the Linux Image this installation is to be associated with.
+     */
+    String imageTag() default "PRIMARY";
```

Comment:
dont we need to tie this with a java tag as well,  so we can generate the correct gradle command etc

Reply:
Yes, you're quite right. I know the Gradle client uses the java set in $JAVA_HOME, but yes, there is the Gradle property `org.gradle.java.home` that can be set for Gradle to use for the build process. I'll add the tag with the idea to use it when setting this property for Gradle.